src/Components/OATPropertyEditor/Internal/PropertyList/PropertyList.tsx

```diff
@@ -170,9 +183,25 @@ const PropertyList: React.FC<IPropertyListProps> = (props) => {
                     property.name
                 );
                 updatedContents[index].name = args.name;
-                updateModel({
+                const updatedModel = {
                     ...selectedItem,
                     contents: updatedContents
+                };
+
+                // validate changes
+                validateOntology({
+                    existingModels: oatPageState.currentOntologyModels,
+                    selectedModelId: oatPageState.selection?.modelId,
+                    originalItem: selectedItem,
+                    updatedItem: updatedModel
+                }).then((validationResult) => {
+                    if (validationResult.isValid === true) {
```

Comment:
same comment here, why not just `if (validationResult.isValid)`

Reply:
same as above, the discriminated union barks without the explicit check.